datafusion/physical-optimizer/src/pushdown_sort.rs

```diff
@@ -0,0 +1,128 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Sort Pushdown Optimization
+//!
+//! This optimizer attempts to push sort requirements down through the execution plan
+//! tree to data sources that can natively handle them (e.g., by scanning files in
+//! reverse order).
+//!
+//! ## How it works
+//!
+//! 1. Detects `SortExec` nodes in the plan
+//! 2. Calls `try_pushdown_sort()` on the input to recursively push the sort requirement
+//! 3. Each node type defines its own pushdown behavior:
+//!    - **Transparent nodes** (CoalesceBatchesExec, RepartitionExec, etc.) delegate to
+//!      their children and wrap the result
+//!    - **Data sources** (DataSourceExec) check if they can optimize for the ordering
+//!    - **Blocking nodes** return `Unsupported` to stop pushdown
+//! 4. Based on the result:
+//!    - `Exact`: Remove the Sort operator (data source guarantees perfect ordering)
+//!    - `Inexact`: Keep Sort but use optimized input (enables early termination for TopK)
+//!    - `Unsupported`: No change
+//!
+//! ## Current capabilities (Phase 1)
+//!
+//! - Reverse scan optimization: when required sort is the reverse of the data source's
+//!   natural ordering, enable reverse scanning (reading row groups in reverse order)
+//! - Supports prefix matching: if data has ordering [A DESC, B ASC] and query needs
+//!   [A ASC], reversing gives [A ASC, B DESC] which satisfies the requirement
+//!
+//! ## Future enhancements (Phase 2)
+//!
+//! - File reordering based on statistics
+//! - Return `Exact` when files are known to be perfectly sorted
+//! - Complete Sort elimination when ordering is guaranteed
+
+use crate::PhysicalOptimizerRule;
+use datafusion_common::config::ConfigOptions;
+use datafusion_common::tree_node::{Transformed, TransformedResult, TreeNode};
+use datafusion_common::Result;
+use datafusion_physical_plan::sorts::sort::SortExec;
+use datafusion_physical_plan::ExecutionPlan;
+use datafusion_physical_plan::SortOrderPushdownResult;
+use std::sync::Arc;
+
+/// A PhysicalOptimizerRule that attempts to push down sort requirements to data sources.
+///
+/// See module-level documentation for details.
+#[derive(Debug, Clone, Default)]
+pub struct PushdownSort;
+
+impl PushdownSort {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl PhysicalOptimizerRule for PushdownSort {
+    fn optimize(
+        &self,
+        plan: Arc<dyn ExecutionPlan>,
+        config: &ConfigOptions,
+    ) -> Result<Arc<dyn ExecutionPlan>> {
+        // Check if sort pushdown optimization is enabled
+        if !config.execution.parquet.enable_sort_pushdown {
```

Comment:
This rule is not parquet specific (though I do realize parquet is the only file source that uses it) so I think it makes sense to make this a normal [optimizer](https://datafusion.apache.org/user-guide/configs.html) flag 

perhaps something like


datafusion.optimizer.enable_window_limits |  
-- | --


<br class="Apple-interchange-newline">